src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs

```diff
@@ -364,7 +343,7 @@ private async Task DownloadServerVersion()
             if (settings.ContainsKey("sonar.cs.msbuild.testProjectPattern"))
             {
                 var value = settings["sonar.cs.msbuild.testProjectPattern"];
```

Comment:
I'd extract the `sonar.cs.msbuild.testProjectPattern` as a constant. It's needed 4 times. And will make the `logWarning` go below 200 characters to fix the code smell